### ISSUE TITLE
🚀 Feature Request: Please fill in the title, don't leave it blank

### ISSUE THREAD

**Issue by hakan-akgul** [tisfeng/Easydict#607]
### Please confirm the following:

- [X] I have carefully read the [README](https://github.com/tisfeng/Easydict#readme)
- [X] I have searched through the [issues](https://github.com/tisfeng/Easydict/issues) page and didn't find a similar feature request.(Including issue that has been closed)
- [X] Easydict has been upgraded to the [latest version](https://github.com/tisfeng/Easydict/releases)
- [X] I understand and agree to the above, and understand that the project maintainer has limited energy, **issues that do not follow the rules may be ignored or closed directly**

### Feature description

Hello everyone. 

Thank you so much for the great app. I love it and I suggested it to my friends.  

Every day I use English, Turkish, and German. So I need to switch between three languages. However, it is hard to find in the long list.  

On the other hand, they are not alphabetically sorted in English.  

It would be great to have a favorite language support in the settings.  
So when we open the translate window we can only list:  
- Auto
- First Language
- Second Language
- Favorite 1
- Favorite 2
- ...  

I have already searched the issues, there was an issue about the "[last languages on top](https://github.com/tisfeng/Easydict/issues/396)". But I think this feature is more suitable.

Thank you all again. Have a nice day.

### Use case

Every day I use English, Turkish, and German. So I need to switch between three languages. However, it is hard to find in the long list.  

### Proposed solution (optional)

It would be great to have a favorite language support in the settings.  
So when we open the translate window we can only list:  
- Auto
- First Language
- Second Language
- Favorite 1
- Favorite 2
- ...  

### Are you willing to submit a PR to implement this feature

- [ ] I'm willing to submit PR

**Comment by tisfeng** [comment]
Good suggestion, adding a favorite language option would be a really nice touch, thanks for the feedback 😊

**Comment by hakan-akgul** [comment]
> Good suggestion, adding a favorite language option would be a really nice touch, thanks for the feedback 😊

I'm glad to hear that. I'm looking forward to the developments. Thank you.

**Comment by tisfeng** [comment]
@AkaShark Can you take this?

**Comment by AkaShark** [comment]
> @AkaShark Can you take this? 你能接受这个吗？

Sure, I will do it. I will be free this weekend.

**Comment by typoman** [comment]
It would be great to trim all the languages to only a small set, not everyone uses all these languages and the list is very long for switching.

**Comment by hakan-akgul** [comment]
> It would be great to trim all the languages to only a small set, not everyone uses all these languages and the list is very long for switching.

At least, it would be better to sort alpabetically. I am lose while searching my native language 😄

**Comment by AkaShark** [comment]
I apologize for being preoccupied with something earlier. I will check this issue soon and break down this feature into three separate features.
1. In the settings, the user can select up to two favorite languages. When the user selects a language, the favorites are displayed at the top of the languages list like
- Auto
- First Language
- Second Language
- Favorite 1
- Favorite 2
...

2. Add a search field to find a language by name in the dropdown menu.
3. Add information about language service support for each language.

For the third feature,  I'm unsure about the display. Adding the information after each language may lead to excessive content in each cell. Does anyone have a good idea for this display?

For the third feature, I'm unsure about the display. Adding the information after each language may lead to excessive content in each cell. Does anyone have a good idea for this?

**Comment by tisfeng** [comment]
> Add information about language service support for each language.

If it's not convenient to display here, we can show the languages they support later on the Settings - Services page.

> Do I need to do this after https://github.com/tisfeng/Easydict/pull/575 or I can start trying recently?

This can be started now. We can add a 'Favorite Languages' option as an entry point in the Settings - General page.

**Comment by tisfeng** [comment]
> In the settings, the user can select up to two favorite languages. When the user selects a language, the favorites are displayed at the top of the languages list like

No, it's not like that. The 'Favorite Languages' is actually a list of commonly used languages; it is an array where users can select multiple languages as their Favorite Languages.

> Every day I use English, Turkish, and German. So I need to switch between three languages. However, it is hard to find in the long list.

"Favorite Languages" is designed to solve this problem. In the query window, when the user clicks on the language selection, we should prioritize displaying "Favorite Languages" to make it convenient for users to quickly choose. Other languages will be placed in the submenu.

<img width="448" alt="image" src="https://github.com/user-attachments/assets/82c9f5c9-14ac-4f96-b4e2-39d72bfbb805">

**Comment by tisfeng** [comment]
"Favorite Languages" may cause ambiguity, it is recommended to use "Common Languages" instead.

This is a UI design for reference, we can make corresponding modifications according to our needs.

<img width="924" alt="image" src="https://github.com/user-attachments/assets/11b1a0b2-cf38-4a9b-9a0b-26658ef94e6c">

**Comment by AkaShark** [comment]
@tisfeng Thanks for your explanation. I got it, and I will continue to push this issue.

**Comment by why-trv** [comment]
Any progress on this?

**Comment by AkaShark** [comment]
> Any progress on this?

Sorry, I've been busy lately. I'll take some time to look at it.

**Comment by dmitrybabich** [comment]
Would it help if someone submitted a PR? This is a showstopper for me since I live in Germany and need to translate from my native language to German and English many times per day, which is very difficult to select from  the list. Also, it would be nice to assign shortcuts for favorite languages, like CMD + 1, CMD + 2, and so on. But that's not critical. Pinning common languages to the top would already help a lot.

**Comment by tisfeng** [comment]
Sorry, we've been busy recently and can't work on this issue for now. 

We welcome anyone to submit a PR.

**Comment by dmitrybabich** [comment]
If it's too time-consuming to add a UI for favoriting languages, would it be possible to just store 2-3 of the most recently used ones and show them on top? Would it be less complicated to implement?
